File: app/src/main/java/com/moka/framework/util/SharedPreferenceManager.java
package com.moka.framework.util;


import android.content.Context;
import android.content.SharedPreferences;


public class SharedPreferenceManager {

	private final String SHARED_PREFERENCE_NAME;

	public SharedPreferenceManager(String sharedPreferenceName) {
		this.SHARED_PREFERENCE_NAME = sharedPreferenceName;
	}

	protected SharedPreferences.Editor getEditor(Context context) {
		return getSharedPreferences(context).edit();
	}

	protected SharedPreferences getSharedPreferences(Context context) {
		return context.getSharedPreferences(SHARED_PREFERENCE_NAME, Context.MODE_PRIVATE);
	}

	public void clearPreference(Context context) {
		getEditor(context).clear().commit();
	}

}

File: app/src/main/java/com/moka/framework/util/SharedPreferenceManager.kt
package com.moka.framework.util


import android.content.Context
import android.content.SharedPreferences


open class SharedPreferenceManager(private val SHARED_PREFERENCE_NAME: String) {

    protected fun getEditor(context: Context): SharedPreferences.Editor {
        return getSharedPreferences(context).edit()
    }

    protected fun getSharedPreferences(context: Context): SharedPreferences {
        return context.getSharedPreferences(SHARED_PREFERENCE_NAME, Context.MODE_PRIVATE)
    }

    fun clearPreference(context: Context) {
        getEditor(context).clear().commit()
    }

}
